ntbs-service/wwwroot/source/Components/AutocompleteSelect.ts

```diff
@@ -45,30 +40,29 @@ const AutocompleteSelect = Vue.extend({
                 this.$emit("input-changed", event);
             }
         },
-        onConfirmValidate: function (value: any) {
-            // this is called twice, value is undefined when user clicks away from input field which is when we want validation to happen
-            if (!this.$props.validate || value != undefined) {
-                return;
-            }
-            this.setSelectValue(this.autocompleteElement.value);
-        },
-        onConfirmFilter: function (value: any) {
-            // this is called twice, value is defined when user makes a selection which is when we want filtering to happen
-            if (!this.$props.filter) {
+        onConfirm: function (value: any) {
+            // this is called twice, when a user selects a value (value is defined) and when user clicks away from input (blur, value undefined)
+            if (!this.$props.emitChangeOn || (this.$props.emitChangeOn == "blur" && value != undefined)) {
                 return;
             }
             if (value != undefined) {
                 this.setSelectValue(value);
             }
+            else if (this.$props.emitChangeOn == "blur") {
+                this.setSelectValue(this.autocompleteElement.value);
+            }

```

Comment:
I think this logic would be clearer without the guard clause, i.e.
```
if (this.$props.emitChangeOn === "blur" && value === undefined) {
    this.setSelectValue(this.autocompleteElement.value);
} else if (this.$props.emitChangeOn === "select" && value !== undefined) {
    this.setSelectValue(value);
}
```

Reply:
It's good form to use `===` rather than `==` for javascript comparisons. Otherwise it's possible for edge cases to sneak in, due to things like the fact that `null == undefined` is `true`.